skbio/sequence/_grammared_sequence.py

```diff
@@ -22,11 +22,79 @@
 from ._sequence import Sequence
 
 
-class IUPACSequence(with_metaclass(ABCMeta, Sequence)):
-    """Store biological sequence data conforming to the IUPAC character set.
+class GrammaredSequenceException(TypeError):
+    pass
+
+
+class GrammaredSequenceMeta(ABCMeta, type):
+    def __new__(mcs, name, bases, dct):
+        cls = super(GrammaredSequenceMeta, mcs).__new__(mcs, name, bases, dct)
+
+        # Only perform validation on classes that aren't abstract.
+        if (type(cls.default_gap_char) is not abstractproperty and
```

Comment:
> If any member is static

@anderspitman I'm assuming when you say "static" you're meaning "abstract", please correct me if I'm wrong.

I don't see how the ABC errors and GrammaredSequenceMeta errors could be interleaved because the ABC errors will happen at instantiation time and the GrammaredSequenceMeta errors will happen at class definition time.

I'm fine with having a single "abstract" guard that only performs validations if all properties are concrete. The more granular approach to validations is nice because the user will know immediately (at class definition time) if there's something wrong with their alphabet, even if some pieces are still abstract. I agree that it may not be worth the complexity and we can always refactor to provide these validations if there's a need in the future.

Are you good with updating the code to only perform validations when all properties are concrete?